### PR TITLE
Test reference inputs build

### PR DESCRIPTION
**Pull request by saratomaz**

### CODE REVIEW

File: cardano_node_tests/tests/test_plutus_v2_spend_build.py [hunk 1]
```diff
@@ -1462,3 +1498,464 @@ def test_v1_attached_v2_reference(
             )
         err_str = str(excinfo.value)
         assert "ReferenceInputsNotSupported" in err_str, err_str
+
+
+@pytest.mark.testnets
+class TestReferenceInputs:
```
Comment (mkoura) [inline cardano_node_tests/tests/test_plutus_v2_spend_build.py:1504]: Make it clear in test class / test function names and docstrings that these tests are for **readonly** reference txins (i.e. txins that will be added to script context), not for reference txins that are actually used for spending.

File: cardano_node_tests/tests/test_plutus_v2_spend_build.py [hunk 2]
```diff
@@ -1462,3 +1498,464 @@ def test_v1_attached_v2_reference(
...
+        )
+        cluster.submit_tx(
+            tx_file=tx_signed, txins=[t.txins[0] for t in tx_output_redeem.script_txins if t.txins]
+        )
+
+        # check that the input used also as reference was spent
+        reference_txin = f"{reference_input[0].utxo_hash}#{reference_input[0].utxo_ix}"
+        reference_input_utxo = cluster.get_utxo(txin=reference_txin)
+
+        assert not reference_input_utxo, f"The reference input was spent `{reference_txin}`"
```
Comment (mkoura) [inline cardano_node_tests/tests/test_plutus_v2_spend_build.py:1698]: `"The reference input was NOT spent...`

File: cardano_node_tests/tests/test_plutus_v2_spend_build.py [hunk 3]
```diff
@@ -1462,3 +1498,464 @@ def test_v1_attached_v2_reference(
...
+                tx_files=tx_files_redeem,
+                readonly_reference_txins=[reference_input],
+                txouts=txouts_redeem,
+                script_txins=plutus_txins,
+            )
+        err_str = str(excinfo.value)
+        assert "TranslationLogicMissingInput" in err_str, err_str
+
+    @allure.link(helpers.get_vcs_link())
+    def test_reference_spent_output(
```
Comment (mkoura) [inline cardano_node_tests/tests/test_plutus_v2_spend_build.py:1780]: How does this test differ from `test_invalid_reference_input`? Both seem to work with non-existent UTxO as txin.

**Review by mkoura — CHANGES_REQUESTED** [review]

File: cardano_node_tests/tests/test_plutus_v2_spend_build.py (hunk 3 above)
Comment (saratomaz) [reply cardano_node_tests/tests/test_plutus_v2_spend_build.py:1780]: Yeah when I was writing that made sense, the first one with a utxo that never exists the second one with a utxo that existed but was already spent. But you are totally right, in the end is the same :) I will drop the first one.

**Review by mkoura — CHANGES_REQUESTED** [review]

**Review by mkoura — APPROVED** [review]

**Pull request merged**